owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/UpdateRemoteShareOperation.kt

```diff
@@ -151,21 +154,29 @@ class UpdateRemoteShareOperation
             putMethod.setRequestBody(formBodyBuilder.build())
 
             putMethod.setRequestHeader(HttpConstants.CONTENT_TYPE_HEADER, HttpConstants.CONTENT_TYPE_URLENCODED_UTF8)
-            putMethod.addRequestHeader(RemoteOperation.OCS_API_HEADER, RemoteOperation.OCS_API_HEADER_VALUE)
+            putMethod.addRequestHeader(OCS_API_HEADER, OCS_API_HEADER_VALUE)
 
             val status = client.executeHttpMethod(putMethod)
 
-            if (isSuccess(status)) {
-                // Parse xml response
-                val parser = ShareToRemoteOperationResultParser(
-                    ShareXMLParser()
-                )
-                parser.ownCloudVersion = client.ownCloudVersion
-                parser.serverBaseUri = client.baseUri
-                result = parser.parse(putMethod.responseBodyAsString)
+            if(!isSuccess(status)){
+                return RemoteOperationResult(putMethod)
+            }
 
-            } else {
-                result = RemoteOperationResult(putMethod)
+            // Parse xml response
+            val parser = ShareToRemoteOperationResultParser(
+                ShareXMLParser()
+            )
+            parser.ownCloudVersion = client.ownCloudVersion
+            parser.serverBaseUri = client.baseUri
+            result = parser.parse(putMethod.responseBodyAsString)
+
+            if (result.isSuccess && retrieveShareDetails) {
+                // retrieve more info - PUT only returns the index of the new share
+                val emptyShare = result.data.shares[0]
```

Comment:
Use first function better
`val emptyShare = result.data.shares.first()`

BTW, I am assuming you are checking shares is not empty, aren't you?

Reply:
Not really, the server returns the updated share with almost no information so I need to retrieve the share with all the information using another operation called GetRemoteShareOperation